src/lib/errors/ApiError.test.ts

```diff
@@ -0,0 +1,74 @@
+import { AxiosResponse } from 'axios';
+
+import ApiError from './ApiErrors';
+
+const mockAxiosResponse: AxiosResponse = {
+  config: {},
+  status: 400,
+  statusText: '',
+  headers: { 'content-type': 'application/json' },
+  data: {
+    errors: [
+      {
+        code: 'internal_error',
+        detail: 'Something went wrong. Please try again.',
+      },
+    ],
+  },
+};
+
+describe('ApiError', () => {
+  it('initializes the ApiError instance', () => {
+    const apiError = new ApiError(mockAxiosResponse);
+
+    expect(apiError.status).toBe(mockAxiosResponse.status);
+    expect(apiError.headers).toBe(mockAxiosResponse.headers);
+    expect(apiError.errors).toBe(mockAxiosResponse.data.errors);
+  });
+
+  describe('.toString()', () => {
+    describe('given there is only one error', () => {
+      it('returns an error message', () => {
+        const axiosResponse = {
+          ...mockAxiosResponse,
+          data: {
+            errors: [
+              {
+                code: 'invalid_email_or_password',
+                detail: 'Your email or password is incorrect. Please try again.',
+              },
+            ],
+          },
+        };
+
+        const apiError = new ApiError(axiosResponse);
+
+        expect(apiError.toString().join(',')).toBe('Your email or password is incorrect. Please try again.');
```

Comment:
We should assert the real value this method is returning and avoid manipulating the results as much as possible 👍 

Then it causes the least confusion as its clear what the expected outcome is 😄 

```suggestion
        expect(apiError.toString()).toBe(['Your email or password is incorrect. Please try again.']);
```

Reply:
But then I'm a bit confused, is the `.join(',')` supposed to be inside the `.toString()` method there?

Reply:
I've changed it to `expect(apiError.toString()[0]).toBe('Your email or password is incorrect. Please try again.');`
and for the to array it should be toArrayString because I return the array of the error 
<img width="818" alt="image" src="https://user-images.githubusercontent.com/122268989/220805590-8a97c7ae-56bb-4d98-be61-91a571206481.png">